### PR TITLE
feat(Steps): new component

### LINKED ISSUE #251: [Feat]: New Steps component
**Description**: The Steps component displays a series of steps that guide users through a process or workflow. It visually indicates progress, helping users understand where they are within a sequence of actions or stages, and what is left to complete.

**Requirements**:

1. **Component Implementation:**

- Implement the Steps component in TypeScript.
- Utilize SCSS with BEM methodology for styling.
- Integrate design tokens from @geneui/tokens for consistent colors, font sizes, spacing, and borders.
- Support horizontal and vertical orientations, with customizable step icons and labels.
- Include options for marking steps as completed, in progress, or uncompleted.
- Allow for optional step descriptions or extra information.
- Provide accessibility features such as ARIA attributes to indicate step progression.

2. **Unit Tests:**

- Write unit tests using Jest and React Testing Library.
- Ensure test coverage for all props, including rendering of different step types (completed, in progress, uncompleted).
- Test for edge cases such as missing labels or unsupported configuration options.

3. **Storybook Stories:**

- Create a Storybook story for the Steps component.
- Showcase different configurations (e.g., horizontal, vertical) and states (e.g., completed, in-progress, uncompleted).
- Create controls for the props so users can interact with the Steps component in Storybook.
- Document all props of the public interfaces.

4. **Figma Design:**

Follow the design outlined in the provided [Figma link.](https://www.figma.com/design/XmW63z2fXfHBC95eCM36ed/Gene-UI%3A-Components?node-id=141-41&p=f&t=OhwlVYVAXp5qQ1tj-0)

**Checklist**:

- [x] Component is implemented in TypeScript.
- [x] Styling follows SCSS using BEM methodology.
- [x] Design tokens from @geneui/tokens are used for colors, font sizes, spacing, and borders.
- [x] Unit tests written and cover different states and edge cases.
- [x] Storybook stories created with examples of all props and step configurations.
- [x] Figma design matches the component implementation.

### PR DESCRIPTION
**Pull request by GaroGabrielyan**
### Chromatic Build Summary 
**Build URL:** https://www.chromatic.com/build?appId=66daee7859a843dd87bfb3d7&number=296 
**Storybook URL:** https://66daee7859a843dd87bfb3d7-xftbhkrvgt.chromatic.com/

### CODE REVIEW

File: src/components/molecules/Steps/Steps.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+     */
+    isLinear?: boolean;
+    /**
+     * Fires when the user interact with Step label. Provides the Step id as a callback's argument.
+     */
+    onChange?: (e: string | number) => void;
+    /**
+     * Loading state for Steps.
+     */
+    isLoading?: boolean;
```
Comment (hamikhambardzumyan) [inline src/components/molecules/Steps/Steps.tsx]: Please remove these two props `isLoading` and `disabled`, as there are no self-responsibilities for representation or functional

File: src/components/molecules/Steps/Steps.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+ * Step component is used to guide users through a sequential process by breaking it down into distinct steps. It is commonly employed in multi-step forms, checkout processes, or workflows that require users to complete tasks in a specific order.
+ */
+const Steps: FC<IStepsProps> = ({
+    direction = "horizontal",
+    type,
+    isLinear,
+    className,
+    children,
+    onChange,
+    isLoading,
```
Comment (hamikhambardzumyan) [inline src/components/molecules/Steps/Steps.tsx]: Remove props `isLoading` and `disabled`

**Review by hamikhambardzumyan — APPROVED** [review]
Looks good to me
